js/entity_browser.common.js

```diff
@@ -20,11 +20,46 @@
    *   Array of selected entities.
    */
   Drupal.entityBrowser.selectionCompleted = function(event, uuid, entities) {
+    var added_entities_array = $.map(entities, function(item) {return item[0]});
+    // @todo Use uuid here. But for this to work we need to move eb uuid
+    // generation from display to eb directly. When we do this, we can change
+    // \Drupal\entity_browser\Plugin\Field\FieldWidget\EntityReference::formElement
+    // also.
+    var cardinality = parseInt(drupalSettings['entity_browser']['field_settings'][uuid]['cardinality']);
+
+    // Having more elements than cardinality should never happen, because
+    // server side authentication should prevent it, but we handle it here
+    // anyway.
+    if (cardinality != -1 && added_entities_array.length > cardinality) {
+      added_entities_array.splice(0, added_entities_array.length - cardinality);
+    }
+
     // Update value form element with new entity IDs.
-    var entity_ids = $(this).parent().parent().find('input[type*=hidden]').val().split(" ");
-    var selected_entity_ids = $.map(entities, function(item) {return item[0]});
-    var entity_ids = _.union(entity_ids, selected_entity_ids);
-    $(this).parent().parent().find('input[type*=hidden]').val(entity_ids.join(" "));
+    var entity_ids = $(this).parent().parent().find('input[type*=hidden]').val();
```

Comment:
You might look into using parents() or closest() here. See https://api.jquery.com/closest/ or https://api.jquery.com/parents/